var/spack/repos/builtin/packages/postgresql-client/package.py

```diff
@@ -0,0 +1,105 @@
+    """PostgreSQL is a powerful, open source object-relational database
+    system.  It has more than 15 years of active development and a
+    proven architecture that has earned it a strong reputation for
+    reliability, data integrity, and correctness. This package comprises
+    only the client libraries and bindings.
+    """
+
+    homepage = "http://www.postgresql.org/"
+    url      = "http://ftp.postgresql.org/pub/source/v9.6.6/postgresql-9.6.6.tar.bz2"
+
+    version('10.3', '506498796a314c549388cafb3d5c717a')
+    version('10.2', 'e97c3cc72bdf661441f29069299b260a')
+    version('9.6.6', '7c65858172597de7937efd88f208969b')
+    version('9.5.3', '3f0c388566c688c82b01a0edf1e6b7a0')
+    version('9.3.4', 'd0a41f54c377b2d2fab4a003b0dac762')
+
+    variant('threadsafe', default=False, description='Enable thread safety.')
+    variant('lineedit',
+            default='readline',
+            values=('readline', 'libedit', 'none'),
+            multi=False,
+            description='Line editing library')
+    variant('python', default=False,
+            description='Enable Python bindings.')
+    variant('perl', default=False,
+            description='Enable Perl bindings.')
+    variant('tcl', default=False,
+            description='Enable Tcl bindings.')
+    variant('gssapi', default=False,
+            description='Build with GSSAPI functionality.')
+
+    provides('postgresql-c')
+    provides('postgresql-python', when='+python')
+    provides('postgresql-perl', when='+perl')
+    provides('postgresql-tcl', when='+tcl')
+
+    depends_on('readline', when='lineedit=readline')
+    depends_on('libedit', when='lineedit=libedit')
+    depends_on('openssl')
+    depends_on('tcl', when='+tcl')
+    depends_on('perl', when='+perl')
+    depends_on('python', when='+python')
+
+    def configure_args(self):
+        config_args = ["--with-openssl"]
+
+        if '+threadsafe' in self.spec:
+            config_args.append('--enable-thread-safety')
+        else:
+            config_args.append('--disable-thread-safety')
+
+        if self.spec.variants['lineedit'].value == 'libedit':
+            config_args.append('--with-libedit-preferred')
+        elif self.spec.variants['lineedit'].value == 'none':
+            config_args.append('--without-readline')
+
+        if '+gssapi' in self.spec:
+            config_args.append('--with-gssapi')
+
+        if '+python' in self.spec:
+            config_args.append('--with-python')
+
+        if '+perl' in self.spec:
+            config_args.append('--with-perl')
+
+        if '+tcl' in self.spec:
+            config_args.append('--with-tcl')
+
+        return config_args
+
+    def install(self, spec, prefix):
+        """Install only the client libraries.
+        """
+        for subdir in ('src/bin', 'src/include', 'src/interfaces', 'src/pl'):
+            with working_dir(join_path(self.build_directory, subdir)):
+                inspect.getmodule(self).make(*self.install_targets)
```

Comment:
You can replace this with:
```python
with working_dir(subdir):
    make()
```
A lot of that cruft is generic stuff from `AutotoolsPackage` that you don't need to add.

Reply:
Huh. Tried as written. Also tried `with working_dir(join_path(self.build_directory, subdir))` as variations. Each attempt ends with:  
```console
==> Executing phase: 'install'
==> Error: InstallError: Install failed for postgresql-client.  Nothing was installed!
InstallError: InstallError: Install failed for postgresql-client.  Nothing was installed!

/home/greenc/work/cet-is/sources/spack/lib/spack/spack/package.py:1673, in sanity_check_prefix:
     19                spack.store.layout.hidden_file_paths)
     20            if not installed:
     21                raise InstallError(
  >> 22                    "Install failed for %s.  Nothing was installed!" % self.name)
```
?

Reply:
Oh yes, sorry, I forgot that this was in install. You can use `make('install')` to fix this.